### ISSUE TITLE
Load exceptions in user resolvers are not reported

### ISSUE THREAD

**Issue by patrick-steele-idem** [benmosher/eslint-plugin-import#1023]
If there is an exception when loading a user resolver then the exception is not reported. For example, if there is a syntax error in a user resolver or if the user has a bad import then the exception is swallowed.

This problem is because of the `try...catch` around the `require()` calls: https://github.com/benmosher/eslint-plugin-import/blob/master/utils/resolve.js#L140-L155

It is much safer to use code that checks for the existence of a module using `require.resolve()`:

```javascript
function tryRequire(target) {
  let resolved;
  try {
    // Check if the target exists
    resolved = require.resolve(target);
  } catch(e) {
    // If the target does not exist then just return undefined
    return undefined;
  }

  // If the target exists then return the loaded module
  return require(resolved);
}
```

I will be submitting a PR.

**Comment by ljharb** [comment]
Fixed in #1024
